src/main/java/spireMapOverhaul/zones/gremlinTown/patches/HordePatches.java

```diff
@@ -116,7 +116,7 @@ public static void Prefix() {
     public static class backAttackPatch {
         @SpirePostfixPatch
         public static void Postfix() {
-            if (AbstractDungeon.lastCombatMetricKey.equals(GremlinTown.GREMLIN_HORDE)) {
+            if (AbstractDungeon.lastCombatMetricKey != null && AbstractDungeon.lastCombatMetricKey.equals(GremlinTown.GREMLIN_HORDE)) {
```

Comment:
No need to change it but just for future reference, you could invert this to be 
`GremlinTown.GREMLIN_HORDE.equals(AD.lastCombatMetricKey)` the constant can never be null and equals(null) is valid. That way you'd save the != null check